docs/sphinx/user_guide/feature/resource.rst

```diff
@@ -0,0 +1,266 @@
+added.
+
+The Resource type that is passed to a ``RAJA::forall`` call must be a concrete type. This is to
+allow compile time assertion that the resource is compatible with the given execution policy. For
+example::
+    
+    using ExecPol = RAJA::cuda_exec_async<BLOCK_SIZE>;
+    RAJA::resources::Cuda my_cuda_res;
+    RAJA::resources::Resource my_res{RAJA::resources::Cuda()};
+    RAJA::resources::Host my_host_res;
+
+    RAJA::forall<ExecPol>(my_cuda_res, .... ) // Compiles.
+    RAJA::forall<ExecPol>(my_res, .... ) // Compilation Error. Not Concrete.
+    RAJA::forall<ExecPol>(my_host_res, .... ) // Compilation Error. Mismatched Resource and Exec Policy.
+
+Below is a list of the current concrete resource types and their execution policy suport.
+
+ ======== ==============================
+ Resource Policy Type
+ ======== ==============================
+ Cuda     | cuda_exec
+          | cuda_exec_async
+ Hip      | hip_exec
+          | hip_exec_async
+ Omp*     | omp_target_parallel_for_exec
+          | omp_target_parallel_for_exec_n
+ Host     | loop_exec
+          | seq_exec
+          | openmp_parallel_exec
+          | omp_for_schedule_exec
+          | omp_for_nowait_schedule_exec
+          | simd_exec
+          | tbb_for_dynamic
+          | tbb_for_static
+ ======== ==============================
+
+.. note:: * The ``RAJA::resources::Omp`` resource is still under development.
+
+IndexSet policies require two execution policies to define them. Currently, a users only need to pass a
+single resource to the forall Indexset call. This resource will be used to execute the inner 
+execution loop of the indexset policy.::
+
+    using ExecPol = RAJA::ExecPolicy<RAJA::seq_segit, RAJA::cuda_exec<256>>;
+    RAJA::forall<ExecPol>(my_cuda_res, iset,  .... );
+
+When a resource is not provided by the user, a *default* resource is assigned from within the RAJA
+forall implementation. This default resource can be accessed in a number of ways.
+
+Directly from the concrete resource type::
+
+    RAJA::resources::Cuda my_default_cuda = RAJA::resources::Cuda::get_defualt();
+
+The resource type can be deduced from an execution policy::
+
+    using Res = RAJA::resources::get_resource<ExecPol>::type;
+    Res r = Res::get_defualt();
+
+Deduced from an execution policy and return the default directly::
+
+    auto my_resource = RAJA::resources::get_default_resource<ExecPol>();
+
+.. note:: * For Cuda and Hip the default resource is *NOT* the CUDA or HIP default stream it is it's 
+            own stream defined in ``camp/include/resource/``. This is in an attempt to break away
+            from some of the issues that arise from the synchronization behaviour of the CUDA/HIP 
+            default stream. It is still possible to use the CUDA/HIP defined default stream as the
+            default resource. This can be enabled by defining ``CAMP_USE_PLATFORM_DEFAULT_STREAM``
```

Comment:
Is this a configure time cmake option, or something I would #define before including RAJA/camp?

Reply:
Would #define before including RAJA, I added a note on this.